tests/api-diff/type_tests.t

```diff
@@ -68,6 +68,8 @@ Run api-watcher on the two cmi files, there should be a difference
 
   $ api-diff ref.cmi modify_type.cmi
   diff module Modify_type:
+  -val f : t -> string
```

Comment:
Is that `t` the one we report a change for below?

Shouldn't it be expanded to `int` and `float` respectively? Or this not fixed by this PR?

If it's indeed not fixed we should add a mention of it in the tests saying that it exhibits a buggy behaviour!

Reply:
This is also not related to this PR and therefore something we should fix separately but we should take care in not printing the value before the type as it is dependent on it! It would be worth opening an issue about that!

Reply:
Yes it should be expanded. But `Ctype.full_expand` doesn't appear to recurse over the type expr, so we have to recurse ourselves, I think it would be better to handle that when we start diffing the core type.

Reply:
Agreed!

Reply:
Could you still add a mention the test is not exhibiting the expected behaviour please?